build.gradle

```diff
@@ -26,7 +26,7 @@ subprojects {
     dependencies {
         testCompile 'junit:junit:4.12',
                 'com.google.guava:guava:18.0',
-                'org.mockito:mockito-core:1.10.19',
+                'org.mockito:mockito-core:3.6.28',
```

Comment:
Cool, I like the upgrade. Just a Q, whats your thoughts on upgrading Junit to 5 too? Your experience with this change, worth doing soon?

Reply:
Yeah I think we should do it, but probably not in this Pr else itll get huge?

Reply:
Yep, lets do as a separate PR